src/ViewsEmprestimo/TelaCadastroEmprestimo.java

```diff
@@ -577,9 +582,16 @@ public ArrayList<ToolModel> getToolsList() {
     private void btnCadastrarEmprestimoActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_btnCadastrarEmprestimoActionPerformed
         if (selectedFriend != null) {
             try {
+                String endDateString = textDataDevolucao.getText();
+                if (endDateString != null && endDateString.isEmpty()) {

```

Comment:
Aqui seria interessante verificar se a data e valida 
por exemplo:

if (endDateString == null ||  (endDateString != null && endDateString.getText().length() != 10)){
           JOptionPane.showMessageDialog(this, "Digite uma data valida (dia/mes/ano)", "Erro", JOptionPane.ERROR_MESSAGE);
           return;
}